tests.py

```diff
@@ -1,29 +0,0 @@
-#!/usr/bin/env python
```

Comment:
Please do not add unrelated changes to large PRs like this one.

Apart from whether this should be removed or not, any unrelated change makes reviewing harder, which may delay the review process, or even discourage reviewers from taking on the PR.

Instead, please keep the PR as focused as possible and submit everything else as a separate PR.

Reply:
These changes were simply to allow me to more easily test my changes (other than using docker, I only have Python 2.7 and 3.8 on my system). I created PR #16986 which cherry picks these changes.

Reply:
Sorry, I guess this comment more belongs on tox.ini the change on this file was that I wasted a lot of time trying to figure why this test failed, and I didn't realize there was a diff file that was generated. I updated this test util to make it more obvious, but I can also pull this out to a different PR

Reply:
Actually this comment is on a deleted file, I will pull this and the change I was talking about ^^^ into a different PR.

Reply:
It also looks like I was wrong about deleting this file. It looks like it expects you have already installed matplotlib in pip editable mode, and I'll suggest that change instead of deleting this file (I looked at its commit log and it has been updated recently so I looked a bit closer)